src/app/Main/Body/QueryPage/chooseSelectedDisplFeatures.service.ts

```diff
@@ -1,9 +1,9 @@
 import { FilterType } from "../../../Shared/Model/Filter.model";
 
 export function chooseSelectedDisplayedFeaturesService(
-    selectedOption: {value: string, label: string}
+    selectedOption: { value: string; label: string } | null
 ): FilterType {
-    if (selectedOption !== undefined && selectedOption.value !== undefined) {
+    if (selectedOption && selectedOption.value) {

```

Comment:
General consideration: prefer always using the exact value you're checking against. E.g. `if(selectedOption.value)` is different from `if(selectedOption.value !== undefined)`. The first check will be false for `value = undefined | null | ""`